extensions/desktop/common/desktop-exports

```diff
@@ -1,11 +1,27 @@
-#!/bin/bash -e
-
+#!/bin/bash
 ###############################################
 # Launcher common exports for any desktop app #
 ###############################################
 
-# shellcheck disable=SC1091 source=extensions/desktop-common/desktop-utils.sh
-source "$(dirname "$0")/desktop-utils.sh"
+function prepend_dir() {
+  local var="$1"
+  local dir="$2"
+  if [ -d "$dir" ]; then
+    eval "export $var=\"\$dir\${$var:+:\$$var}\""
+  fi
+}
+
+function append_dir() {
+  local var="$1"
+  local dir="$2"
+  if [ -d "$dir" ]; then
+    eval "export $var=\"\${$var:+\$$var:}\$dir\""
+  fi
+}
+
+function can_open_file() {
+  head -c0 "$1" &> /dev/null
```

Comment:
How about `[ -f "$1" ] && [ -r "$1" ]` ?

Reply:
I believe this is due to the fact that `test -f` and `test -r` doesn't reveal whether the file is readable after applying Apparmor security confinement.